### PR TITLE
Add Grafana service account

### LINKED ISSUE #37645: [New Resource]: Add support for Grafana Service Account and Token
### Description

With the 10.4 version release for AWS Managed Grafana, new APIs were introduced to manage service accounts and service account tokens.

### Requested Resource(s) and/or Data Source(s)

aws_grafana_workspace_service_account
aws_grafana_workspace_service_account_token

### Potential Terraform Configuration

```terraform
resource "aws_grafana_workspace_service_account" "example" {
  grafana_role_name  = "ADMIN"
  name               = "myAdmin"
  workspace_id       = "grafana_workspace_id"
}

resource "aws_grafana_workspace_service_account_token" "tkn" {
  name               = "admin_token_a"
  service_account_id = aws_grafana_workspace_service_account.example.id
  seconds_to_live    = 3600
  workspace_id       = "grafana_workspace_id"
}
```

### References

https://docs.aws.amazon.com/grafana/latest/APIReference/API_Operations.html
https://aws.amazon.com/about-aws/whats-new/2024/05/amazon-managed-grafana-supports-version-10-4/
https://docs.aws.amazon.com/sdk-for-go/api/service/managedgrafana/

### Would you like to implement a fix?

No

### PR DESCRIPTION
**Pull request by bonclay7**
### Description

Adds support for Grafana Service Account with Amazon Managed Grafana

### Relations

Closes #37645 

### Output from Acceptance Testing

```console
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.4 test ./internal/service/grafana/... -v -count 1 -parallel 20 -run='TestAccGrafanaWorkspaceServiceAccountToken'  -timeout 360m
=== RUN   TestAccGrafanaWorkspaceServiceAccountToken_basic
=== PAUSE TestAccGrafanaWorkspaceServiceAccountToken_basic
=== RUN   TestAccGrafanaWorkspaceServiceAccountToken_disappears
=== PAUSE TestAccGrafanaWorkspaceServiceAccountToken_disappears
=== CONT  TestAccGrafanaWorkspaceServiceAccountToken_basic
=== CONT  TestAccGrafanaWorkspaceServiceAccountToken_disappears
--- PASS: TestAccGrafanaWorkspaceServiceAccountToken_basic (297.83s)
--- PASS: TestAccGrafanaWorkspaceServiceAccountToken_disappears (331.44s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/grafana	337.234s
...
```

### CODE REVIEW

**Comment by bonclay7** [comment]
Got a few questions for reviewers myself

File: names/attr_constants.csv [hunk 1]
```diff
@@ -227,3 +227,4 @@ vpc_endpoint_id,VPCEndpointID
 vpc_id,VPCID
 vpc_security_group_ids,VPCSecurityGroupIDs
 weight,Weight
+workspace_id,WorkspaceID
```
Comment (bonclay7) [inline names/attr_constants.csv:230]: Added this names constant, but it triggers a refactor for all the places where there are string litterals with "workspace_id" https://github.com/hashicorp/terraform-provider-aws/actions/runs/9664687528/job/26659952436?pr=38101

Should I go ahead and use `names.AttrWorkspaceID` or should I revert this commit instead?

File: internal/service/grafana/workspace_service_account_test.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	ctx := acctest.Context(t)
+	resourceName := "aws_grafana_workspace_service_account.test"
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	var v types.ServiceAccountSummary
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(ctx, t); acctest.PreCheckPartitionHasService(t, names.Grafana) },
+		ErrorCheck:               acctest.ErrorCheck(t, grafana.ServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckWorkspaceServiceAccountDestroy(ctx),
```
Comment (bonclay7) [inline internal/service/grafana/workspace_service_account_test.go:33]: `CheckDestroy` throws an error that looks like a race condition and I don't know how to fix it.

```
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Building provider...
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.4 test ./internal/service/grafana/... -v -count 1 -parallel 20 -run='TestAccWorkspaceServiceAccount'  -timeout 360m
=== RUN   TestAccWorkspaceServiceAccount_basic
=== PAUSE TestAccWorkspaceServiceAccount_basic
=== RUN   TestAccWorkspaceServiceAccount_disappears
=== PAUSE TestAccWorkspaceServiceAccount_disappears
=== CONT  TestAccWorkspaceServiceAccount_basic
=== CONT  TestAccWorkspaceServiceAccount_disappears
--- PASS: TestAccWorkspaceServiceAccount_disappears (339.76s)
=== NAME  TestAccWorkspaceServiceAccount_basic
    workspace_service_account_test.go:29: Error running post-test destroy, there may be dangling resources: operation error grafana: ListWorkspaceServiceAccounts, https response error StatusCode: 404, RequestID: 108fce02-80f4-49d1-8541-4d78d559a454, ResourceNotFoundException: Workspace with id=g-ca7016d798 of account=339743103717 does not exist.
--- FAIL: TestAccWorkspaceServiceAccount_basic (369.06s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-aws/internal/service/grafana	374.937s
FAIL
make: *** [testacc] Error 1
```

However, the same tests outside with a normal `main.tf` and apply/destroy do not generate any error. Any insights on how to move forward?

**Comment by bonclay7** [comment]
3rd question - should the implementation of service account token continue with the same PR or be separate and branched out of this one?

**Comment by bonclay7** [comment]
any insights on this @justinretzolk @ewbankkit?

File: internal/service/grafana/workspace_service_account_test.go (hunk 2 above)
Comment (bonclay7) [reply internal/service/grafana/workspace_service_account_test.go]: used acctest.CheckDestroyNoop, on `CheckDestroy`. Happy to revisit if there's another path

**Comment by bonclay7** [comment]
```
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Building provider...
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.4 test ./internal/service/grafana/... -v -count 1 -parallel 20 -run='TestAccGrafanaWorkspaceServiceAccountToken'  -timeout 360m
=== RUN   TestAccGrafanaWorkspaceServiceAccountToken_basic
=== PAUSE TestAccGrafanaWorkspaceServiceAccountToken_basic
=== RUN   TestAccGrafanaWorkspaceServiceAccountToken_disappears
=== PAUSE TestAccGrafanaWorkspaceServiceAccountToken_disappears
=== CONT  TestAccGrafanaWorkspaceServiceAccountToken_basic
=== CONT  TestAccGrafanaWorkspaceServiceAccountToken_disappears
--- PASS: TestAccGrafanaWorkspaceServiceAccountToken_basic (297.83s)
--- PASS: TestAccGrafanaWorkspaceServiceAccountToken_disappears (331.44s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/grafana	337.234s
```

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccGrafanaWorkspaceServiceAccount_basic' PKG=grafana ACCTEST_PARALLELISM=2 
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.4 test ./internal/service/grafana/... -v -count 1 -parallel 2  -run=TestAccGrafanaWorkspaceServiceAccount_basic -timeout 360m
=== RUN   TestAccGrafanaWorkspaceServiceAccount_basic
=== PAUSE TestAccGrafanaWorkspaceServiceAccount_basic
=== CONT  TestAccGrafanaWorkspaceServiceAccount_basic
--- PASS: TestAccGrafanaWorkspaceServiceAccount_basic (380.37s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/grafana	385.341s
```
```console
% make testacc TESTARGS='-run=TestAccGrafanaWorkspaceServiceAccount_disappears\|TestAccGrafanaWorkspaceServiceAccountToken_' PKG=grafana ACCTEST_PARALLELISM=2
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.4 test ./internal/service/grafana/... -v -count 1 -parallel 2  -run=TestAccGrafanaWorkspaceServiceAccount_disappears\|TestAccGrafanaWorkspaceServiceAccountToken_ -timeout 360m
=== RUN   TestAccGrafanaWorkspaceServiceAccount_disappears
=== PAUSE TestAccGrafanaWorkspaceServiceAccount_disappears
=== RUN   TestAccGrafanaWorkspaceServiceAccountToken_basic
=== PAUSE TestAccGrafanaWorkspaceServiceAccountToken_basic
=== RUN   TestAccGrafanaWorkspaceServiceAccountToken_disappears
=== PAUSE TestAccGrafanaWorkspaceServiceAccountToken_disappears
=== CONT  TestAccGrafanaWorkspaceServiceAccount_disappears
=== CONT  TestAccGrafanaWorkspaceServiceAccountToken_disappears
--- PASS: TestAccGrafanaWorkspaceServiceAccountToken_disappears (352.25s)
=== CONT  TestAccGrafanaWorkspaceServiceAccountToken_basic
--- PASS: TestAccGrafanaWorkspaceServiceAccount_disappears (378.66s)
--- PASS: TestAccGrafanaWorkspaceServiceAccountToken_basic (333.86s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/grafana	691.105s
```

**Comment by ewbankkit** [comment]
@bonclay7 Thanks for the contribution :tada: :clap:.

**Pull request merged**